hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/TracingFilter.java

```diff
@@ -71,7 +71,7 @@ public void filter(ContainerRequestContext requestContext,
     // written the data to the destination
     OutputStream out = responseContext.getEntityStream();
     if (out != null) {
-      responseContext.setEntityStream(new FilterOutputStream(out) {
+      responseContext.setEntityStream(new PureWrapperOutputStream(out) {
```

Comment:
It makes sense to wrap the outputStream only  if tracing is enabled, By default tracing is disabled 
```java 
public static final boolean HDDS_TRACING_ENABLED_DEFAULT = false;
```
So this wrapping won't be of any help if disabled since tracer registration happens only if the above property is true (Please verify this part) , We should check if Tracing is enabled , only then do the wrap. If it is disabled, do nothing.

Reply:
Thank you @sadanand48 for the review, made the changes as suggested!